File: transformed-array.ts
{
    function constructTransformedArray(nums: number[]): number[] {
        const result: number[] = [];
        const n = nums.length;

        for (let i = 0; i < n; i++) {
            if (nums[i] !== 0) {
                let newIndex = ((i + nums[i]) % n + n) % n;  
                result.push(nums[newIndex]);
            } else {
                result.push(0);
            }
        }
        
        return result;
    };
}

File: transformed-array.rs
pub fn construct_transformed_array(nums: Vec<i32>) -> Vec<i32> {
    let n = nums.len();
    let mut res = vec![0; n];

    for i in 0..n {
        res[i] = nums[((i as i32 + nums[i]) % (n as i32) + (n as i32)) as usize % n];
    }
    return res;
}
